src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -143,6 +143,12 @@ public abstract class CommandLineArguments
         public ImmutableArray<CommandLineSourceFile> AdditionalFiles { get; internal set; }
 
         /// <summary>
+        /// An ordered list of .editorconfig files that are consumed by analyzers. The order is significant, and if more than one .editorconfig
```

Comment:
`DiscoverEditorConfigFiles` does not apply the necessary partial order to its output such that the semantics are correct when files are passed to this method.